### PR TITLE
[UT] add test for pkg/probe/http/request.go

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?
add test for pkg/probe/http/request.go

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/probe/http/request_test.go [hunk 1]
```diff
@@ -38,3 +44,53 @@ func TestFormatURL(t *testing.T) {
...
+func Test_userAgent(t *testing.T) {
+	tests := []struct {
+		name    string
+		purpose string
+		want    string
+	}{
+		{
+			name:    "test useragent output",
+			purpose: "agent",
+			want:    "kube-agent/.",
```
Comment (aojea) [inline pkg/probe/http/request_test.go:57]: I'm not sure if this test can flake in the future

```
func userAgent(purpose string) string {
	v := version.Get()
	return fmt.Sprintf("kube-%s/%s.%s", purpose, v.Major, v.Minor)
}

```

it depends on the output of `version.Get` , it seems is returning empty now ... this test doesn't looks worth adding it

File: pkg/probe/http/request_test.go [hunk 2]
```diff
@@ -38,3 +44,53 @@ func TestFormatURL(t *testing.T) {
...
+}
+
+func Test_v1HeaderToHTTPHeader(t *testing.T) {
+	tests := []struct {
+		name       string
+		headerList []v1.HTTPHeader
+		want       http.Header
+	}{
+		{
+			name: "test1",
```
Comment (aojea) [inline pkg/probe/http/request_test.go:76]: can you add a case with an empty input to assert the behavior?

File: pkg/probe/http/request_test.go (hunk 1 above)
Comment (my-git9) [reply pkg/probe/http/request_test.go]: It has been removed, thx

File: pkg/probe/http/request_test.go (hunk 2 above)
Comment (my-git9) [reply pkg/probe/http/request_test.go:56]: Updated.f

Comment (aojea) [reply pkg/probe/http/request_test.go:56]: I mean, another test case, 

```
{
			name: "empty input",

}
```

also, use a more meaningful name for this test , `test1` is not very informative, `common headers` per example, ...

File: pkg/probe/http/request_test.go [hunk 3]
```diff
@@ -38,3 +45,35 @@ func TestFormatURL(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v1HeaderToHTTPHeader(tt.headerList); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("v1HeaderToHTTPHeader() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+
+	gg := v1HeaderToHTTPHeader(nil)
+	assert.NotNil(t, gg)
```
Comment (aojea) [inline pkg/probe/http/request_test.go:78]: this should be a test case in the array of tests

**Comment by bart0sh** [comment]
@my-git9 It doesn't have to be in this PR, but it would be nice to test other functions and uncovered code in the same file.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by derekwaynecarr** [comment]
Thank you for the test.

/lgtm
/approve

**Pull request merged**